### ISSUE TITLE
Status of MLDataPattern porting

### ISSUE THREAD

**Issue by CarloLucibello** [JuliaML/MLUtils.jl#3]
A list of what is currently exported from MLDataPattern.jl.

### TO PORT
- [x] `getobs`, `getobs!` and `nobs`. #1    
    - `nobs` is now `numobs`;
    - `obsdim` argument is dropped from the interface
- [x]  `randobs` #1 
- [x]  `datasubset, DataSubset` #4
- [x] `shuffleobs` #5 
- [x] `splitobs` #5 
- [x] `DataView` #5
   - [x]  Consider removal 
- [x] `obsview, ObsView` #5 
- [x] `batchview, BatchView` #6 
- [x] `batchsize` #6 
- [x] `slidingwindow, SlidingWindow` #196 
- [x] `stratifiedobs` (now `splitobs(..., stratified=y)`, #195)
- [x] `oversample, undersample` #10 
- [x] `kfolds` #9 
- [x] `leaveout` #9 
- [x] `eachobs` #9 
- [x] `eachbatch` #9 

### NOT TO BE PORTED
- `BufferGetObs`
- `RandomObs, RandomBatches`
- `BalancedObs`
- `FoldView`
-  `targets`
-  `eachtarget`

**Comment by CarloLucibello** [comment]
We can consider this essentially done

**Comment by rmkn85** [comment]
Hi, what about `stratifiedobs` and `slidingwindow`?
Were they explicitly excluded on purpose?
Thanks

**Comment by CarloLucibello** [comment]
not really, we just didn't port code that we weren't sure was going to be useful. I think `stratifiedobs` should go in, less sure of `slidingwindow` but didn't look much into it and alternatives in the ecosystem.

**Comment by rmkn85** [comment]
Just to clarify, I came here specifically for missing `stratifiedobs`.
It is needed to replicate the behaviour of Python's `sklearn.model_selection.train_test_split([...] stratify=true)`

Asked about `slidingwindow` on the way, since it was the only other one unchecked but not in the list of explicitly "not to be ported".

**Comment by kevindirect** [comment]
I use `slidingwindow` often for time series data. Haven't looked too much for a replacement but the closest I've found is [IterTools.jl partition](https://juliacollections.github.io/IterTools.jl/stable/#). It has a similar interface but returns a tuple iterator

**Comment by kevindirect** [comment]
Also found `DSP.Periodograms.arraysplit` which is similar to `slidingwindow` but you set the overlap instead of the stride. So far `slidingwindow` is the fastest of the three because it returns views.

**Comment by CarloLucibello** [comment]
`stratified` split has been implemented in #195
